### ISSUE TITLE
How can I add the math library in the project.yml for a build with the coverage plugin?

### ISSUE THREAD

**Issue by stemschmidt** [ThrowTheSwitch/Ceedling#857]
Hi @mvandervoord, I have to use the gcov plugin for test coverage reports. How can I provide the linker flag -lm for the build for gcov and where is this documented?
Best regards

**Comment by mkarlesky** [comment]
Hi, @stemschmidt. You've got two options here.

1. The official way is to use the `:libraries` project configuration. This area of Ceedling's configuration area is specific to library handling and includes a number of options.
1. A simple, convenient hack is to add a flag to your `gcov` plugin linker's command line just for linking the math library.

The following examples assume the common convention of a math library named `libm`, referenced without the `lib` by the GNU linker's command line using the `-l` flag.

```
:libraries:
  :test:
    - m
```

```
:tools_gcov_linker:
  :arguments:
    - lm
```

(Incidentally, we're nearing releasing Ceedling 0.32. The documentation is _vastly_ improved. An addition I am yet working on is a Ceedling cookbook to gather together common needs and how to use Ceedling's features to meet them. `:libraries` has been documented forever. The “hack” I mentioned has only recently been officially documented in the pre-release of 0.32.)

**Comment by stemschmidt** [comment]
Hi Michael, that was quick! Thanks a lot for the reply. I am struggling to find patterns in the project.yml file. I understand that a lot is configurable, but if there is no clear description of what is available it is very frustrating to get things running. Especially if time is limited. Thanks for pointing me to the :tools_gcov_linker: section, I think I haven't seen it anywhere in the documentation.

**Comment by mkarlesky** [comment]
Appreciate that feedback. Creating easy-to-use build tools — or any tool for power users — is darn hard. There's so much glue and so many possible needs to support. Invariably the tool becomes complicated and hard to explain in simple terms. A big goal for the upcoming 0.32 release was simply to fill in lots of holes in the documentation and provide many more examples. It's _so_ much more complete. Our next big task is to extract all that documentation from the docs/ repository folder and transform it into something searchable, navigable, etc. As to how best to create documentation or a power tool itself that is easy to grasp… we'd love to be pointed to tools you think are good exemplars. We'll happily borrow the ideas to make Ceedling better!

**Comment by stemschmidt** [comment]
Hi Michael, the "official way" solved my problem. Thanks! Or better, solved that problem :-) 

What I also noticed is that my 
:defines:
  :common: 
.....
which are used in 

  :test:
    - *common_defines
  :test_preprocess:
    - *common_defines

are not forwarded to the gcov compiler. Any idea how I could set those (I am currently setting and clearing CFLAGS in my environment. That is not nice). Can I open a :gcov: section in :defines: ?

**Comment by swaldhoer** [comment]
@stemschmidt are you on `0.31.1` or using the `0.32.*rc`?

- In `0.31.1` the defines are not passed to the gcov build, but only in the normal build.
- In `0.32.*rc` it works as expected and both builds get the defines :)

**Comment by stemschmidt** [comment]
I am on 0.31.1...

**Comment by mkarlesky** [comment]
@stemschmidt Unfortunately, we are not likely to be supporting 0.31.1 with code changes as the next release is soon to be published. We just don't have the resources. As @swaldhoer mentioned, your issue is fixed with the 0.32 pre-release. You could likely hack what you need in 0.31.1 with the same tool trick I mentioned earlier. Only, here it would be `:tools_gcov_compiler`. You could add the needed `-D` arguments to the Gcov build's compilation steps.
